tests/integration/internal/stepfuncs/assess_funcs.go

```diff
@@ -75,3 +77,49 @@ func WaitUntilExpectedMetricsPresent(
 		return ctx
 	}
 }
+
+// WaitUntilExpectedMetricsPresent returns a features.Func that can be used in `Assess` calls.
+// It will wait until the provided number of logs with the provided labels are returned by receiver-mock's HTTP API on
+// the provided Service and port, until it succeeds or waitDuration passes.
+func WaitUntilExpectedLogsPresent(
+	expectedLogsCount uint,
+	expectedLogsMetadata map[string]string,
+	receiverMockNamespace string,
+	receiverMockServiceName string,
+	receiverMockServicePort int,
+	waitDuration time.Duration,
+	tickDuration time.Duration,
+) features.Func {
+	return func(ctx context.Context, t *testing.T, envConf *envconf.Config) context.Context {
+		kubectlOpts := *ctxopts.KubectlOptions(ctx)
+		kubectlOpts.Namespace = receiverMockNamespace
+		terrak8s.WaitUntilServiceAvailable(t, &kubectlOpts, receiverMockServiceName, int(waitDuration), tickDuration)
+
+		client, closeTunnelFunc := receivermock.NewClientWithK8sTunnel(ctx, t)
+		defer closeTunnelFunc()
+
+		assert.Eventually(t, func() bool {
+			logsCount, err := client.GetLogsCount(t, expectedLogsMetadata)
```

Comment:
If you do this in `assert.Eventually` and this fails once (marking `t` as failed) then the whole test will fail regardless of the final result. IIRC

Reply:
True, but it should only fail in extreme circumstances (like if we get a 200 and can't decode the data, for example). Some errors should cause the whole test to fail, and I think it's easier if the function itself decides this instead of having to inspect the error here.